mvpa2/generators/resampling.py

```diff
@@ -203,3 +203,74 @@ def generate(self, ds):
     def __str__(self):
         return _str(self, str(self._amount), n=self._attr, count=self.count,
                     apply_selection=self._apply_selection)
+
+
+class NonContiguous(Node):
+    """Generator to remove samples too close as measured with a sample attribute
+    ``` partitioner = ChainNode([NFoldPartitioner(),
```

Comment:
We use numpy style docstrings, and run doctests which allows to assure that code we inject is correct.  Ideally such a snippet should be entered in the same fashion, see various places around the code (where there is Examples in docstrings) .e.g  within mvpa2/base/dataset.py